### PR TITLE
Add switches tab to system inventory

### LINKED ISSUE #2556: Add switches tab to inventory page
Some design in https://github.com/oxidecomputer/console/issues/1451 but it's just a table with columns for  whatever is on the switch response.

https://docs.oxide.computer/api/switch_list

https://github.com/oxidecomputer/console/blob/510c6457cc7fd18c1706e206dd07369e1aaa3cd3/app/api/__generated__/Api.ts#L3349-L3359

https://github.com/oxidecomputer/console/blob/510c6457cc7fd18c1706e206dd07369e1aaa3cd3/app/api/__generated__/Api.ts#L231-L234

### PR DESCRIPTION
**Pull request by charliepark**
This adds a tab to the inventory page, showing switches.

Screenshot 2024-12-12 at 10 31 07 AM

Closes #2556

### CODE REVIEW

File: app/pages/system/inventory/SwitchesTab.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export async function loader() {
+  await queryClient.prefetchQuery(switchList.optionsFn())
+  return null
+}
+
+const colHelper = createColumnHelper<Switch>()
+const staticCols = [
+  colHelper.accessor('id', {}),
+  colHelper.accessor('rackId', { header: 'Rack ID' }),
```
Comment (david-crespo) [inline app/pages/system/inventory/SwitchesTab.tsx:34]: I think we can drop rack ID until we have multirack. It's going to be the same one meaningless UUID for everything — sleds, disks, etc.

File: app/pages/system/inventory/SwitchesTab.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  colHelper.accessor('baseboard.part', { header: 'part number' }),
+  colHelper.accessor('baseboard.serial', { header: 'serial number' }),
+  colHelper.accessor('baseboard.revision', { header: 'revision' }),
```
Comment (david-crespo) [inline app/pages/system/inventory/SwitchesTab.tsx:37]: If this is useful here, it's probably useful on the other tabs. Worth asking someone.

**Comment by benjaminleonard** [comment]
Wish we had some identifiers that were a little more human-friendly here. But not something we can solve at the console-level right?

**Comment by david-crespo** [comment]
Yeah, not really anything we can do.

**Pull request merged**